src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -325,8 +338,10 @@ public TextDocumentState GetAdditionalDocumentState(DocumentId documentId)
             ImmutableDictionary<DocumentId, DocumentState> documentStates = null,
             ImmutableDictionary<DocumentId, TextDocumentState> additionalDocumentStates = null,
             AsyncLazy<VersionStamp> latestDocumentVersion = null,
-            AsyncLazy<VersionStamp> latestDocumentTopLevelChangeVersion = null)
+            AsyncLazy<VersionStamp> latestDocumentTopLevelChangeVersion = null,
+            ValueSource<ProjectStateChecksums> lazyChecksums = null)
         {
+            // for now, re-calculate all checksums when anything changed
```

Comment:
you wrote the comment right above. currently that is the case, I pass along lazyChecksums but at the end, I always assign new AsyncLazy.